regulation-worker/internal/client/client.go

```diff
@@ -164,12 +164,15 @@ func prepURL(url string, params ...string) string {
 }
 
 func mapPayloadToJob(wjs jobSchema, workspaceID string) (model.Job, error) {
-	usrAttribute := make([]model.UserAttribute, len(wjs.UserAttributes))
-	for i := 0; i < len(wjs.UserAttributes); i++ {
-		usrAttribute[i] = model.UserAttribute{
-			UserID: wjs.UserAttributes[i].UserID,
-			Phone:  wjs.UserAttributes[i].Phone,
-			Email:  wjs.UserAttributes[i].Email,
+	usrAttribute := make([]model.User, len(wjs.UserAttributes))
+	for i, usrAttr := range wjs.UserAttributes {
+		usrAttribute[i].Attributes = make(map[string]string)
+		for key, value := range usrAttr {
+			if key == "userId" {
```

Comment:
how do we guarantee that `userId` field will always be present in the response? Or is it acceptable to have a `User` with an empty `ID`?

A test scenario would be desirable to disclose the desired system behaviour if this happens

Reply:
That's a good point. `userId` is a required field. Although the reason we don't have a check for it is, regulation-manager throws error if any user facing API request doesn't have userId. So, we assumed in regulation-worker that regulation-manager itself will take care of it. So, we don't really need to worry about it.

Reply:
I am personally confused, if we should have a redundant check in regulation-worker or not?

One line of thought is, since it is checked at regulation-manager so it doesn't makes sense to check again here.
But another line of thought is, we shouldn't depend of regulation-manager (a different component) to make sure that it will send appropriate request with all the fields as per our requirement. 

But, at the same time, everything in itself could of thought of as a component, so redundantly who many times should we do the same check again & again. As in, the client here passes `job` to `service` (another component) which further passes job to `deleter` (another component). So, Where do we draw the line?

Reply:
I guess we could filter a response entry if userID is not present in it. But if we are OK to fail when sending the request to the service and we handle it gracefully, I' m fine with that too.